gcc/rust/backend/rust-compile-context.h

```diff
@@ -222,11 +222,20 @@ class TyTyResolveCompile : public TyTy::TyVisitor
 
   virtual ~TyTyResolveCompile () {}
 
-  void visit (TyTy::ErrorType &type) override { gcc_unreachable (); }
+  void visit (TyTy::ErrorType &type ATTRIBUTE_UNUSED) override
```

Comment:
Apparently a more portable means of preventing the warning is to simply omit the parameter name.

For example, visiting the ErrorType would be written as so:
```
void visit (TyTy::ErrorType &) override
{
  gcc_unreachable ();
}
```

I have been told that this is more portable (something about attributes not being interpreted in the same way with different C++ compilers) but I don't actually know whether this is truly the case or not.

Reply:
I think your right it looks really ugly too see the unused attributed like that too.